mapie/quantile_regression.py

```diff
@@ -316,13 +323,82 @@ def _check_cv(
                 "Invalid cv method, only valid method is ``split``."
             )
 
-    def fit(  # type: ignore
+    def _check_calib_set(
         self,
         X: ArrayLike,
         y: ArrayLike,
-        X_calib: ArrayLike,
-        y_calib: ArrayLike,
         sample_weight: Optional[ArrayLike] = None,
+        X_calib: Optional[ArrayLike] = None,
+        y_calib: Optional[ArrayLike] = None,
+        calib_size: Optional[float] = 0.3,
+        random_state: Optional[Union[int, np.random.RandomState, None]] = None,
+        shuffle: Optional[bool] = True,
+        stratify: Optional[ArrayLike] = None,
+    ) -> Tuple[
+        ArrayLike, ArrayLike, ArrayLike, ArrayLike, Optional[ArrayLike]
+    ]:
+        """
+        HPUT SOMETHING
+
+        Parameters
+        ----------
+        Same definition of parameters as for the ``fit`` method.
+
+        Returns
+        -------
+        Tuple[ArrayLike, ArrayLike, ArrayLike, ArrayLike, ArrayLike]
+            - [0]: ArrayLike, X_train
```

Comment:
Add the shape ?

Reply:
done
e.g. 
for `X_train`: `[0]: ArrayLike of shape (n_samples_*(1-calib_size), n_features)`
or 
for `X_calib`: `[2]: ArrayLike of shape (n_samples_*calib_size, n_features)`

Is that the correct method?

Reply:
Yes perfect !